### ISSUE TITLE
PromiseLike.then returns `PromiseLike<Promise<T>>` instead of `PromiseLike<T>`

### ISSUE THREAD

**Issue by andy-ms** [microsoft/TypeScript#17862]
**TypeScript Version:**  nightly (2.6.0-dev.20170817)

**Code**

```ts
declare const p: PromiseLike<string>;
const x: PromiseLike<number> = p.then(() => Promise.resolve(1));
```

**Expected behavior:**

No error.

**Actual behavior:**

```
src/a.ts(2,7): error TS2322: Type 'PromiseLike<Promise<number>>' is not assignable to type 'PromiseLike<number>'.
  Type 'Promise<number>' is not assignable to type 'number'.
```

Discovered in `DefinitelyTyped/types/xadesjs`.

**Comment by spion** [comment]
FYI this is now also happening in mainline TS (`2.6.2`)

**Comment by jakebailey** [comment]
Old issue, but this was fixed back in #40002.
